Reconcile "components" and refactor to use Forms

jonahtwalker opened the pull request:
<!-- This template is to serve as a comprehensive outline for documenting updates to the code. Some of these sections may not be relevant to your PR. In that case, document it inline with `N/A`  -->

## What does this PR do?
<!-- Provide a clear and concise description of what this PR accomplishes -->

### Related Issues & PRs
<!-- List the related issues, using either: `Closes #123` or `Fixes #123` , which will automatically move the specified issue to "In Review" when the PR is created -->

#### Key Decisions
<!-- Explain the technical decisions and alternative approaches considered for your new implementations and refactored solutions -->
- **Implemeneted Approach**:
  <!-- What did you decide on? -->
- **Alternatives Considered**:
  <!-- What else did you consider? -->
- **Rationale**:
  <!-- Why did you decide this was the most appropriate solution? -->

## Breaking Changes
<!-- Do others need to do something to successfully incorporate your changes into their work? -->
- [ ] This PR includes breaking changes
  <!-- What do they need to do? -->

### Limitations & Concerns
<!-- Describe any limitations, potential edge cases, or situations where your solution might break -->
- **Known limitations**:
- **Edge cases**:
- **Potential risks**:

---

### UI Changes
<!-- If this PR modifies the UI, provide screenshots and descriptions -->
- [ ] This PR includes UI changes
<!-- Include screenshots in the table below, include before & after applicable -->

### Screenshots
| Description | Screenshot |
|-------------|------------|
| Before      | <!-- img(s) --> |
| After       | <!-- img(s) --> |

---

### Reviewers
<!-- Tag at least 2 others for review -->
- [ ] Reviewer 1: @username
- [ ] Reviewer 2: @username

### Next Steps
<!-- What needs to happen after this PR is merged? -->
- [ ] Follow-up tasks are created
  - [ ] Task 1

### How to Test
<!-- Provide detailed steps for QA to test the changes -->
1. Prerequisites:
  <!-- e.g. Do I need to be a specific role? -->
2. Test steps:
   <!-- What is the quickest path to testing your solution? -->
3. Expected results:
   <!-- Is there any nuance that should be clarified? If so, can the implementation get more simple instead? -->

## Final Checklist
- [ ] Linked related issues
- [ ] Rebased with target branch
- [ ] No lint errors
- [ ] QA'd in multiple browsers
- [ ] Added/updated tests (if applicable)
- [ ] Mobile responsive (if applicable)
- [ ] Updated documentation (if applicable)
- [ ] Notified the Team of this updates readiness

---

## AI Usage
<!-- Document any AI tools you used to develop this -->
- [ ] AI tools were used in development
  - **Tool used**:
    <!-- Specify any model used -->
  - **Prompts used**:
    <!-- Format these inclusions as quotes, like: `> Provide  -->
    - **Initial prompt**: 
- [ ] No AI tools were used
  - **Reason**:
    <!-- Why not? -->

## References
<!-- List any helpful references used during development -->
- **Documentation**:
- **Discussions**:
- **Related PRs**:

jonahtwalker committed:
- Refactoring to make use of forms

jonahtwalker committed:
- refactored signin

jonahtwalker committed:
- updates
- removed default form actions

jonahtwalker committed:
Remove context import

jonahtwalker committed:
- fixed identtype

jonahtwalker committed:
- fixed reset password to work with username
- schema normalization

jonahtwalker committed:
- Refactoring Sign Up, UX incomplete

rupalirajput2597 commented:
Hello @jonahtwalker , I noticed that the reset password flow is not working correctly:
- Users can reset their password even if they enter incorrect information.
- The "Verify My Email" button sometimes remains disabled, possibly due to the debounce function used in the onChange event within `reset-password-form.svelte.`
sharing screen recording:

https://github.com/user-attachments/assets/9b1977b6-1411-44c8-b68c-d6a420220770

rupalirajput2597 commented on src/routes/reset-password/reset-password-form.svelte:
```diff
@@ -0,0 +1,421 @@
+
+      isValidating = true;
+
+      debounceTimer = window.setTimeout(async () => {
```
Why are we using `debounceTimer` here? It seems to be causing an unnecessary delay.

jonahtwalker replied:
Where is there another debounce in effect? It should try to validate the form when the input is changed.

rupalirajput2597 commented on src/routes/signup/form/sign-up-form2.svelte:
```diff
@@ -0,0 +1,470 @@
+      isValidating = true;
+
+      debounceTimer = window.setTimeout(async () => {
+        try {  
```
Why are we using `debounceTimer` here? It seems to be causing an unnecessary delay.

jonahtwalker replied:
Where is there another debounce in effect? It should try to validate the form when the input is changed.

rupalirajput2597 committed:
RR/refactored-password-input and added unhappy errors for signin and reset-password (#72)

* added password visibility toggle
* added some client error handling in resetpassword and signup
* formatted code

---------

Co-authored-by: Rupali Rajput <rupali@Mac-7809.lan>

jonahtwalker commented:
/gemini review

jonahtwalker committed:
- More squashing and sign in changes
- added error styling to inputs
- sign up needs to be hooked up and cleaned
- needs better client message handling, to invalidate invalid inputs

jonahtwalker committed:
- reset password advances to step 3 before we know token validity
- probably going to ditch the listener handler, thought it'd be nicer
- lots of logging to clean up

jonahtwalker committed:
- updated reset password
- reinstated listener

jonahtwalker committed:
- infer resetpassword schema change

jonahtwalker committed:
- more normalizing in sign in

jonahtwalker committed:
Fix the removeListener helper

jonahtwalker committed:
- Made sign up form operational
